lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchDriverConfiguration.java

```diff
@@ -115,6 +144,13 @@
       .bindNamedParameter(AzureStorageAccountName.class, AZURE_STORAGE_ACCOUNT_NAME)
       .bindNamedParameter(AzureStorageContainerName.class, AZURE_STORAGE_CONTAINER_NAME)
 
+      // Bind Azure Container Parameters
+      .bindNamedParameter(ContainerRegistryServer.class, CONTAINER_REGISTRY_SERVER)
+      .bindNamedParameter(ContainerRegistryUsername.class, CONTAINER_REGISTRY_USERNAME)
+      .bindNamedParameter(ContainerRegistryPassword.class, CONTAINER_REGISTRY_PASSWORD)
+      .bindNamedParameter(ContainerImageName.class, CONTAINER_IMAGE_NAME)
+      .bindNamedParameter(TcpPortListString.class, TCP_PORT_LIST_STRING)
+
```

Comment:
Why are we specifying TcpPortListString and not PortListProvider?

Reply:
If I specify the TcpPortProvider to be ListTcpPortProvider, that will change the default implementation (when containers is not used) to use ListTcpPortProvider and not RangeTcpPortProvider.

AzureBatchDriverConfigurationProviderImpl.getDriverConfiguration() has the logic to bind to ListTcpPortProvider or not based on whether this is a container-based pool or not.

---
In reply to: [196516355](https://github.com/sharathmalladi/reef/pull/5#discussion_r196516355) [](ancestors = 196516355)